src/backend/gporca/libgpopt/src/operators/CPhysicalHashJoin.cpp

```diff
@@ -392,8 +409,8 @@ CPhysicalHashJoin::PdshashedMatching(
 			GPOS_WSZ_LIT("Unable to create matching hashed distribution."));
 	}
 
-	return GPOS_NEW(mp) CDistributionSpecHashed(
-		pdrgpexpr, false /* fNullsCollocated */, opfamilies);
+	return GPOS_NEW(mp)
+		CDistributionSpecHashed(pdrgpexpr, fNullsColocated, opfamilies);
```

Comment:
```suggestion
BOOL fNullsColocated = !FPhysicalOuterJoin();
return GPOS_NEW(mp)
		CDistributionSpecHashed(pdrgpexpr, fNullsColocated, opfamilies);
```

Also remove lines L332~L337.

Reply:
Defined fNullscolocated right before returning the matching spec